.github/workflows/int-and-connected-test-run.yml

```
@@ -26,15 +26,17 @@ jobs:
           cache: 'gradle'
       - name: Run unit tests
         env:
-          GOOGLE_APPLICATION_CREDENTIALS: 'jade-dev-account.json'
+          # required for unit tests
+          GOOGLE_APPLICATION_CREDENTIALS: jade-dev-account.json
+          # required for sonarqube reports
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          # output plain logs instead of json
           TDR_LOG_APPENDER: 'Console-Standard'
         run: |
           # extract service account credentials
           base64 --decode <<< ${{ secrets.SA_B64_CREDENTIALS }} > ${GOOGLE_APPLICATION_CREDENTIALS}
           # assemble code, run unit tests, and generate scan
-          ./gradlew assemble
-          ./gradlew check --scan jacocoTestReport sonar
+          ./gradlew --scan assemble check jacocoTestReport sonar
```

I may have missed the response to an earlier comment, but why do these steps use `assemble`? For this invocation, `check jacocoTestReport sonar` should be sufficient, and the one below could be `testConnected`.
```suggestion
          ./gradlew --scan check jacocoTestReport sonar
```

---

The gradle step dependency tree is not obvious so I think this is clearer, but we can always improve this in the future.

---

I guess the way I think about it is, `check` would be incorrect if it didn't also cause a build to occur. Using `assemble` causes a jar to be built, which is unnecessary as nothing in this step requires the jar.

You can always use `gradle --dry-run <task>` to see all the tasks that a task will invoke.

But it works fine either way can be changed later, I agree.